utils/zerovec/README.md

```diff
@@ -0,0 +1,44 @@
+# zerovec [![crates.io](http://meritbadge.herokuapp.com/zerovec)](https://crates.io/crates/zerovec)
+
+Zero-copy vector abstractions over byte arrays.
+
+This crate has two main types:
+
+- `ZeroVec<T>` for fixed-width types like `u32`
+- `VarZeroVec<T>` for variable-width types like `str`
+
+Both are intended as drop-in replacements for `Vec<T>` in Serde structs serialized with a
+format supporting a borrowed byte buffer, like Bincode.
+
+## Features
+
+- `serde`: enables Serde Serialize/Deserialize impls for ZeroVec and VarZeroVec.
+
+## Example
+
+```rust
+use zerovec::ZeroVec;
+
+// This example requires the "serde" feature
+#[derive(serde::Serialize, serde::Deserialize)]
+pub struct DataStruct<'s> {
+    #[serde(borrow)]
+    nums: ZeroVec<'s, u32>,
+}
+
+let data = DataStruct {
+    nums: ZeroVec::from_aligned(&[211, 281, 421, 461]),
+};
+let bincode_bytes = bincode::serialize(&data)
+    .expect("Serialization should be successful");
+assert_eq!(24, bincode_bytes.len());
+
+let deserialized: DataStruct = bincode::deserialize(&bincode_bytes)
+    .expect("Deserialization should be successful");
+assert_eq!(Some(211), deserialized.nums.first());
+assert!(matches!(deserialized.nums, ZeroVec::Borrowed(_)));
+```
```

Comment:
`request`: Since the value proposition is performance/memory, can you provide `# Performance` section that makes some rough claims?
You can either paste benchmark results or just make a rough claim to allow the reader to evaluate whether they care about the impact.

Reply:
In order to resolve this request, I needed to add benchmarks to VarZeroVec and polish the ones for ZeroVec, which I was planning to do anyway.  I have added those benchmark improvements to this PR and documented the results in lib.rs / README.md.